File: design-patterns/cpp/facade/build_system.cpp
#include "build_system.h"

bool BuildSystem::compileProject() {
    std::cout << "BuildSystem: Compiling project..." << std::endl;
    simulateDelay(2000);
    std::cout << "BuildSystem: Build successful." << std::endl;
    return true;
}

std::string BuildSystem::getArtifactPath() {
    std::string path = "target/myapplication-1.0.jar";
    std::cout << "BuildSystem: Artifact located at " << path << std::endl;
    return path;
}

void BuildSystem::simulateDelay(int ms) {
    std::this_thread::sleep_for(std::chrono::milliseconds(ms));
} 

File: design-patterns/python/facade/build_system.py
import time

class BuildSystem:
    def compile_project(self) -> bool:
        print("BuildSystem: Compiling project...")
        self._simulate_delay(2)
        print("BuildSystem: Build successful.")
        return True

    def get_artifact_path(self) -> str:
        path = "target/myapplication-1.0.jar"
        print(f"BuildSystem: Artifact located at {path}")
        return path
    
    def _simulate_delay(self, seconds: int) -> None:
        time.sleep(seconds) 